src/commands/ProjectCommand.ts

```diff
@@ -0,0 +1,60 @@
+import Command from "../abstracts/Command";
+import {Message, MessageEmbed} from "discord.js";
+import {readFile} from "fs";
+
+export default class ProjectCommand extends Command {
+	private readonly fileDirectory = "./assets/projects.json";
+	constructor() {
+		super(
+			"project",
+			"Returns a random project idea based on given parameters.",
+		);
+	}
+
+	async run(message: Message, args: string[]): Promise<void> {
+		const embed = new MessageEmbed();
+		const query = args.map(arg => arg.toLowerCase()).filter(arg => arg.trim().length > 0);
+
+		if (args.length === 0) {
+			embed.setTitle("Projects")
+				.setColor("#add8e6")
+				.setDescription("Please provide arguments on the projects command.");
+			await message.channel.send(embed);
+		} else {
+			readFile(`${this.fileDirectory}`, {encoding: "utf8"}, async (error, projectData) => {
+				const projects: Array<Project> = JSON.parse(projectData);
+				const project = projects
+					.filter(this.removeTooLongDescriptions)
+					.filter(project => this.filterTags(project, query))
+					.sort(() => 0.5 - Math.random()).pop();
+
+				if (project) {
+					const difficulty = this.retrieveFirstFoundTag(project, ["easy", "medium", "hard"]);
```

Comment:
Thoughts on making a `ProjectDifficulty` enum rather than magic strings?

Reply:
I had considered an enum. However, I decided against it ultimately and decided that a difficulty is ultimately just another tag of the project. This reduced complexity by just having a tag. The easy / medium/ hard thing will just provide a neat border.

Reply:
Hmm okay, that makes sense I suppose. I'm just not sure I like the magic strings - could they at least be constants?